python_modules/dagster/dagster/_core/host_representation/external.py

```diff
@@ -216,7 +272,12 @@ def solid_names_in_topological_order(self):
 
     @property
     def external_pipeline_data(self):
-        return self._external_pipeline_data
+        if self._data is None:
+            if self._ref is None or self._ref_to_data_fn is None:
+                check.failed("unexpected state - unable to load data from ref")
+            self._data = self._ref_to_data_fn(self._ref)
```

Comment:
worried about thread safety here - is there anywhere where two different threads call this on the same object? Before it was immutable

Reply:
maybe this won't cause issues other than possibly duplicated work if its called from two different threads? still spooky

Reply:
ya agree it shouldn't break anything but could cause repeat fetches